xml/src/main/java/org/ehcache/xml/ConfigurationParser.java

```diff
@@ -105,6 +112,23 @@
   private final Map<URI, CacheResourceConfigurationParser> resourceXmlParsers = new HashMap<URI, CacheResourceConfigurationParser>();
   private final ConfigType config;
 
+  static String getReplacementString(String originalValue) {
+    Matcher matcher = SYSPROP.matcher(originalValue);
+
+    StringBuffer sb = new StringBuffer();
+    while (matcher.find()) {
+      final String property = matcher.group(1);
+      final String value = System.getProperty(property);
+      if (value == null) {
+        return null;
```

Comment:
Y... I don't know what the right behavior should be. There is a nasty one currently (that iirc ehcache v2 also suffer of): a failed substitution will create some ugly persistence dir... and i _HATE_ when that happens... Logging doesn't fix it though :(